File: app/src/main/java/codes/wokstym/cookingrecipes/models/RecipeDto.java
package codes.wokstym.cookingrecipes.models;


import java.io.Serializable;
import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RecipeDto implements Serializable {

    private UUID id;
    private String preparation;
    private List<IngredientDto> ingredients;
    private MealTime mealTime;
    private int prepTime;
    private BigDecimal price;
    private String title;
}

File: app/src/main/java/codes/wokstym/cookingrecipes/models/RecipeDto.kt
package codes.wokstym.cookingrecipes.models

import android.os.Parcelable
import kotlinx.parcelize.Parcelize
import java.math.BigDecimal
import java.util.*

@Parcelize
data class RecipeDto(
    val id: UUID,
    val preparation: String,
    val ingredients: List<IngredientDto>,
    val mealTime: MealTime,
    val prepTime: Int,
    val price: BigDecimal,
    val title: String,
    val pictureUrl: String?
) : Comparable<RecipeDto>, Parcelable {

    fun getPreparations(): List<String> = preparation.split("\n").map { it.trim() }

    override fun compareTo(other: RecipeDto): Int {
        if (this.id == other.id)
            return 0

        if (this.mealTime.order != other.mealTime.order)
            return this.mealTime.order - other.mealTime.order

        return this.title.compareTo(other.title)
    }
}